### PR TITLE
Adding systables that list files in database directory

### PR DESCRIPTION
**Pull request by riverszhang89**
This pull request adds 3 system tables `comdb2_files`, `comdb2_log_files` and `comdb2_tmp_files` that list files in the database's main, log and tmp directory, respectively (similar to PG's `pg_ls_dir`, `pg_ls_waldir` and `pg_ls_tmpdir`). For example, the statement below returns the names of llmeta and meta:

```
cdb2sql> SELECT * FROM comdb2_files WHERE file LIKE '%meta%'
(file='comdb2_llmeta.dta')
(file='comdb2_metadata.dta')
[SELECT * FROM comdb2_files WHERE file LIKE '%meta%'] rc 0
```

### CODE REVIEW

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ sqlite/ext/comdb2/tables.c @@
+ If a new system table is added, be sure to document it in docs/pages/programming/system_tables.md.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Error &#x26A0;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>4/472 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sc_transactional_rowlocks_generated<br>userschema_queueodh_generated<br>userschema<br>ddl_analyze</p>

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ sqlite/ext/comdb2/tables.c @@
+ If a new system table is added, be sure to document it in docs/pages/programming/system_tables.md.
```

File: sqlite/ext/comdb2/files.c [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+{
+    char **data;
+    int n;
+    const char *fname;
+    struct dirent *buf, *ent;
+    DIR *dirp = opendir(dir);
+
+    data = NULL;
+    n = 0;
+    dirp = opendir(dir);
```
Comment (chands10) [inline sqlite/ext/comdb2/files.c:48]: Looks like this directory is opened twice (also on line 44)

File: sqlite/ext/comdb2/files.c [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+int ls_tmp(void **datap, int *np)
+{
+    return ls_dir(thedb->bdb_env->tmpdir, datap, np);
+}
+
+void free_ls(void *data, int n)
+{
+    free(data);
```
Comment (chands10) [inline sqlite/ext/comdb2/files.c:86]: I think you need to free each element of data?

**Review by chands10 — CHANGES_REQUESTED** [review]

**Comment by riverszhang89** [comment]
@chands10 Thanks for catching those!

**Comment by riverszhang89** [comment]
@chands10 Thanks for the review! However Nirbhay has a more complete version of this in https://github.com/bloomberg/comdb2/pull/2726. Withdrawing mine ...

**Pull request closed without merging**